datalad/core/distributed/push.py

```diff
@@ -0,0 +1,735 @@
+
+    This makes the last saved state of a dataset available to a sibling
+    or special remote data store of a dataset. Any target sibling must already
+    exist and be known to the dataset.
+
+    Optionally, it is possible to limit a push to change sets relative
+    to a particular point in the version history of a dataset (e.g. a release
+    tag). By default, the state of the local dataset is evaluated against the
+    last known state of the target sibling. An actual push is only attempted
+    if there was a change compared to the reference state, in order to speed up
+    processing of large collections of datasets. Evaluation with respect to
+    a particular "historic" state is only supported in conjunction with a
+    specified reference dataset. Change sets are also evaluated recursively, i.e.
+    only those subdatasets are pushed where a change was recorded that is
+    reflected in the current state of the top-level reference dataset.
+    See "since" option for more information.
+
+    Only a push of saved changes is supported.
+
+    .. note::
+      Power-user info: This command uses :command:`git push`, and :command:`git annex copy`
+      to push a dataset. Publication targets are either configured remote
+      Git repositories, or git-annex special remotes (if they support data
+      upload).
+    """
+
+    # TODO add examples
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc=""""specify the dataset to push""",
+            constraints=EnsureDataset() | EnsureNone()),
+        to=Parameter(
+            args=("--to",),
+            metavar='SIBLING',
+            doc="""name of the target sibling. If no name is given an attempt is
+            made to identify the target based on the dataset's configuration
+            (i.e. a configured tracking branch, or a single sibling that is
+            configured for push)""",
+            constraints=EnsureStr() | EnsureNone()),
+        since=Parameter(
+            args=("--since",),
+            constraints=EnsureStr() | EnsureNone(),
+            doc="""specifies commit-ish (tag, shasum, etc.) from which to look for
+            changes to decide whether pushing is necessary.
+            If an empty string is given, the last state of the current branch
+            at the sibling is taken as a starting point."""),
+        path=Parameter(
+            args=("path",),
+            metavar='PATH',
+            doc="""path to contrain a push to. If given, only
+            data or changes for those paths are considered for a push.""",
+            nargs='*',
+            constraints=EnsureStr() | EnsureNone()),
+        force=Parameter(
+            # multi-mode option https://github.com/datalad/datalad/issues/3414
+            args=("-f", "--force",),
+            doc="""force particular operations, overruling automatic decision
+            making: use --force with git-push ('gitpush'); do not use --fast
```

Comment:
nit-pick: `('gitpush')` is the only option given in brackets in this text that has quotation marks (i.e., no quotes around options here `[...] git-annex copy (datatransfer); [...] file content (no-datatransfer); [...] modes 'gitpush' and 'datatransfer' (all)`)

Reply:
Everything is quoted now.